packages/components/Button/src/FAB/FAB.types.ts

```diff
@@ -1,22 +1,38 @@
 import { ButtonSlotProps, ButtonCoreTokens, ButtonCoreProps } from '../Button.types';
 import { ShadowProps } from '@fluentui-react-native/experimental-shadow';
+import { IconSourcesType } from '@fluentui-react-native/icon';
 
 export const fabName = 'FAB';
 export type FABSize = 'small' | 'large';
+export type FABAppearance = 'primary' | 'subtle' | 'ascent';
 
 export interface FABSlotProps extends ButtonSlotProps {
   shadow?: ShadowProps;
 }
 
 export interface FABProps extends ButtonCoreProps {
+  /**
+   * A FAB can have its content and borders styled for greater emphasis or to be subtle.
+   * - 'primary' or 'ascent': Emphasizes the button as a primary action.
+   *   'ascent' is mobile naming convention, 'primary' included here to maintain parity with Button.
+   * - 'subtle': Minimizes emphasis to blend into the background until hovered or focused.
+   * @default 'primary' (or 'ascent')
+   */
+  appearance?: FABAppearance;
+
+  /*
+   * Source URL or name of the icon to show on the Button. Overrides optionality.
```

Comment:
>Overrides optionality. [](http://example.com/codeflow?start=59&length=22)

Can you clarify what this means? What optionality is being overridden?

Reply:
'icon' exists in ButtonCoreProps as an optional prop. Redeclaring it in FABProps to override its optionality type, didn't want to just have it removed from ButtonCoreProps (into ButtonProps) since it would lead to the question of why 'icon' is it not in the common props. This felt cleaner. Would you prefer we do this differently?